sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineOperationsTests.java

```diff
@@ -104,6 +108,34 @@ protected void cleanUpResources() {
         }
     }
 
+    @Test
+    public void canCreateAndUpdateVirtualMachineWithUserData() {
+        String userDataForCreate = new String(Base64.getEncoder().encode(
+            UUID.randomUUID().toString().toUpperCase(Locale.ROOT).getBytes(StandardCharsets.UTF_8)));
+        String userDataForUpdate = new String(Base64.getEncoder().encode(
+            UUID.randomUUID().toString().toUpperCase(Locale.ROOT).getBytes(StandardCharsets.UTF_8)));
+
+        // Create
+        VirtualMachine vm = computeManager
+            .virtualMachines()
+            .define(vmName)
+            .withRegion(region)
+            .withNewResourceGroup(rgName)
+            .withNewPrimaryNetwork("10.0.0.0/28")
+            .withPrimaryPrivateIPAddressDynamic()
+            .withoutPrimaryPublicIPAddress()
+            .withPopularWindowsImage(KnownWindowsVirtualMachineImage.WINDOWS_SERVER_2012_R2_DATACENTER)
```

Comment:
Can we use a later windows version? 2012 seems too old.

Reply:
1. The value of  `userData` changed to two different non-random strings.
2. The version of windows virtual machine image changed to `2019-datacenter-gensecond`.